spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -293,6 +294,8 @@ public RabbitProducerProperties getExtendedProducerProperties(String channelName
 	protected MessageHandler createProducerMessageHandler(final ProducerDestination producerDestination,
 			ExtendedProducerProperties<RabbitProducerProperties> producerProperties, MessageChannel errorChannel)
 			throws Exception {
+		Assert.state(!producerProperties.getHeaderMode().equals(HeaderMode.embeddedHeaders),
+				"the RabbitMQ binder does not support embedded headers since RabbitMQ supports headers natively");
```

Comment:
How about to follow Postal's Law and just ignore and WARN otherwise fail fast?
It doesn't look like critical option which makes the whole system not working.

Otherwise LGTM.

Reply:
That was my initial thought; but we'd need to change the property value so the abstract binder won't set up for embedded headers. I can do that, but it seems a bit dirty to me - someone might refactor the code in future so the property is read before we get here.

Reply:
Sure! Sounds reasonable.
Merging...